app/main.py

```diff
@@ -1,214 +1,67 @@
-        arthur["protection"] += a["protection"]
-
-    # apply weapon
-    arthur["power"] += arthur["weapon"]["power"]
-
-    # apply potion if exist
-    if arthur["potion"] is not None:
-        if "power" in arthur["potion"]["effect"]:
-            arthur["power"] += arthur["potion"]["effect"]["power"]
-
-        if "protection" in arthur["potion"]["effect"]:
-            arthur["protection"] += arthur["potion"]["effect"]["protection"]
-
-        if "hp" in arthur["potion"]["effect"]:
-            arthur["hp"] += arthur["potion"]["effect"]["hp"]
-
-    # mordred
-    mordred = knightsConfig["mordred"]
-
-    # apply armour
-    mordred["protection"] = 0
-    for a in mordred["armour"]:
-        mordred["protection"] += a["protection"]
-
-    # apply weapon
-    mordred["power"] += mordred["weapon"]["power"]
-
-    # apply potion if exist
-    if mordred["potion"] is not None:
-        if "power" in mordred["potion"]["effect"]:
-            mordred["power"] += mordred["potion"]["effect"]["power"]
-
-        if "protection" in mordred["potion"]["effect"]:
-            mordred["protection"] += mordred["potion"]["effect"]["protection"]
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
+from app.knights.knight import Knight
+from app.knights.describtion import Characteristics
+from app.additional_config.wepon import Weapon
+from app.additional_config.protection import Armour
+from app.additional_config.potion import Potion
+
+
+def battle(dict_knight: dict) -> dict:
+    result_dict = {}
+
+    for knight in dict_knight:
+        fighter = Knight(dict_knight[knight]["name"])
+        basic_config = Characteristics(dict_knight[knight]["hp"],
+                                       dict_knight[knight]["power"])
+        weapon = Weapon(dict_knight[knight]["weapon"])
+        armour = Armour(dict_knight[knight]["armour"])
+        result = basic_config.basic_knights_config(weapon.power_weapon,
+                                                   armour.protection_armour)
+        result_dict.update({fighter.name: result})
+
+        # apply potion if exist
+        if dict_knight[knight]["potion"] is not None:
```

Comment:
u can write just `if dict_knight[knight]["potion"]:` for checking is it None or not